update dhcp.script to handle dynamic routing.

mogulapranay-freescale opened the pull request:
Certain DHCP servers push a gateway outside of the assigned interface subnet,
to support those situations, install a host route towards the gateway.

If Gateway and IP are served in same network, openwrt quagga cannot learn routes (rip routes are
not getting added, showing inactive) whereas working fine when Gateway and IP are in different network.

Signed-off-by: Mogula Pranay <mogula.pranay@nxp.com>

mogulapranay-freescale committed:
update dhcp.script to handle dynamic routing.

Certain DHCP servers push a gateway outside of the assigned interface subnet,
to support those situations, install a host route towards the gateway.

If Gateway and IP are served in same network, openwrt quagga cannot learn routes (rip routes are
not getting added, showing inactive) whereas working fine when Gateway and IP are in different network.

-mogula.pranay@nxp.com

dedeckeh commented on package/network/config/netifd/files/lib/netifd/dhcp.script:
```diff
@@ -20,7 +20,13 @@ setup_interface () {
-		proto_add_ipv4_route "$i" 32 "" "$ip"
+		local ip_net
+		local gw_net
+		ip_net=$(ipcalc.sh $ip/$mask | grep "NETWORK" | awk -F '=' '{print $2}')
```
Usage of grep and awk is awkward; use eval "$(ipcalc.sh $ip/$mask)"; ip_net="$NETWORK".
Bring the ip_net calculation out of the loop as it does not make sense to calculate it for every loop iteration

dedeckeh commented on package/network/config/netifd/files/lib/netifd/dhcp.script:
```diff
@@ -20,7 +20,13 @@ setup_interface () {
+		local ip_net
+		local gw_net
+		ip_net=$(ipcalc.sh $ip/$mask | grep "NETWORK" | awk -F '=' '{print $2}')
+		gw_net=$(ipcalc.sh $i/$mask | grep "NETWORK" | awk -F '=' '{print $2}')
```
Usage of grep and awk is awkward; use eval "$(ipcalc.sh $i/$mask)"; gw_net="$NETWORK".

dedeckeh commented on package/network/config/netifd/files/lib/netifd/dhcp.script:
```diff
@@ -20,7 +20,13 @@ setup_interface () {
+		local gw_net
+		ip_net=$(ipcalc.sh $ip/$mask | grep "NETWORK" | awk -F '=' '{print $2}')
+		gw_net=$(ipcalc.sh $i/$mask | grep "NETWORK" | awk -F '=' '{print $2}')
+		if [ "$ip_net" != "$gw_net" ]; then
```
Use [ "$ip_net" != "$gw_net" ] && proto_add_ipv4_route

dedeckeh reviewed (commented):
Your PR has formal issues:

    missing prefix in commit subject (should be netifd: )
    missing Signed-off-by

Please have a look at the Submitting patches guideline for more details.

Please use git commit --amend to add changes to your commit. Use git push --force to overwrite your github repository/branch. The PR will afterwards update automatically. Please don't close this PR and open a new one.

mogulapranay-freescale committed:
netifd: update dhcp.script to handle dynamic routing.

Certain DHCP servers push a gateway outside of the assigned interface subnet,
to support those situations, install a host route towards the gateway.
If Gateway and IP are served in same network, openwrt quagga cannot learn routes (rip routes are
not getting added, showing inactive) whereas working fine when Gateway and IP are in different network.
Signed-off-by: Mogula Pranay mogula.pranay@nxp.com
